src/main/kotlin/systems/ajax/englishstudytelegrambot/service/WordService.kt

```diff
@@ -1,35 +1,70 @@
 package systems.ajax.englishstudytelegrambot.service
 
-import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
-import systems.ajax.englishstudytelegrambot.exception.WordNotFoundBySpendingExceptions
+import systems.ajax.englishstudytelegrambot.dto.WordDto
 import systems.ajax.englishstudytelegrambot.entity.Word
+import systems.ajax.englishstudytelegrambot.exception.WordAlreadyPresentInLibraryException
+import systems.ajax.englishstudytelegrambot.exception.WordNotFoundBySpendingException
+import systems.ajax.englishstudytelegrambot.repository.LibraryRepository
 import systems.ajax.englishstudytelegrambot.repository.WordRepository
 
 interface WordService {
 
-    suspend fun save(wordSpelling: String, wordTranslate: String): Word
+    suspend fun saveNewWord(libraryName: String, telegramUserId: String, wordDto: WordDto): Word
 
-    fun getWordByItsSpelling(wordSpelling: String): Word
+    fun updateWordTranslate(libraryName: String, telegramUserId: String, wordDto: WordDto): Word
+
+    fun deleteWord(libraryName: String, telegramUserId: String, wordSpelling: String): Word
+
+    fun getFullInfoAboutWord(libraryName: String, telegramUserId: String, wordSpelling: String): Word
 }
 
 @Service
 class WordServiceImpl(
     val wordRepository: WordRepository,
+    val libraryRepository: LibraryRepository,
     val additionalInfoAboutWordService: AdditionalInfoAboutWordService
 ) : WordService {
 
-    override suspend fun save(wordSpelling: String, wordTranslate: String): Word {
-        val word =
-            Word(wordSpelling, wordTranslate, additionalInfoAboutWordService.findAdditionInfoAboutWord(wordSpelling))
-        log.info("full saving word {}", word)
-        return wordRepository.save(word)
+    override suspend fun saveNewWord(libraryName: String, telegramUserId: String, wordDto: WordDto): Word {
+        val libraryId = libraryRepository.getLibraryIdByPairLibraryNameAndTelegramUserId(libraryName, telegramUserId)
+        libraryId.takeIf {
+            !wordRepository.isWordBelongsToLibraryByWordSpelling(wordDto.spelling, libraryId)
+        }?.let {
+            return wordRepository.saveNewWord(
+                Word(
+                    spelling = wordDto.spelling,
+                    translate = wordDto.translate,
+                    libraryId = libraryId,
+                    additionalInfoAboutWord = additionalInfoAboutWordService.findAdditionInfoAboutWord(wordDto.spelling)
+                )
+            )
```

Comment:
```suggestion
           val additionalInfo = additionalInfoAboutWordService.findAdditionInfoAboutWord(wordDto.spelling)
            return wordRepository.saveNewWord(
                Word(
                    spelling = wordDto.spelling,
                    translate = wordDto.translate,
                    libraryId = libraryId,
                    additionalInfoAboutWord = additionalInfo
                )
            )
```
it's better not to call other services when constructing an object